.github/actions/inject-issue-id-action/action.yaml

```diff
@@ -0,0 +1,98 @@
+---
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: 2024 The Linux Foundation
+
+name: "🎟️ Add JIRA ticket to commit message"
+
+inputs:
+  ISSUE_ID_LOOKUP_JSON:
+    description: "JSON mapping users to JIRA tickets"
+    type: string
+    required: true
+  INJECT:
+    # When injection is enabled and no issue found...
+    # String will be added, output variable set true
+    description: "Whether to inject an issue-id if NOT found"
+    type: boolean
+    required: false
+    default: true
+
+outputs:
+  PRESENT:
+    description: "Whether the required ticket string is present"
+    value: ${{ steps.status.outputs.present }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ github.head_ref }}
+
+    - name: "Get GIT commit message body"
+      id: commit-message
+      # yamllint disable rule:line-length
+      uses: lfit/releng-reusable-workflows/.github/actions/git-commit-message-action@main
+
+    - name: "Check commit message for ticket/string"
+      shell: bash
+      run: |
+        # Check commit message for ticket/string
+        echo "# Checking Commit Message for JIRA Ticket" >> "$GITHUB_STEP_SUMMARY"
+
+        COMMIT_MESSAGE="${{ env.COMMIT_MESSAGE }}"
+        if [ -z "$COMMIT_MESSAGE" ]; then
+          echo "Commit message body was empty/unpopulated ❌"; exit 1
+        fi
+        if [[ "$COMMIT_MESSAGE" =~ "Issue-ID: " ]]; then
+          echo "Ticket string found in the pull request ✅" >> "$GITHUB_STEP_SUMMARY"
+          echo "Ticket string found in the pull request ✅"
+          echo "PRESENT=true" >> "$GITHUB_ENV"
+        else
+          echo "Ticket string NOT found in the pull request ❌" >> "$GITHUB_STEP_SUMMARY"
+          echo "Ticket string NOT found in the pull request ❌"
+          echo "PRESENT=false" >> "$GITHUB_ENV"
+        fi
+
+    - name: "Set key to use for JSON lookup"
+      if: env.PRESENT == 'false' && inputs.INJECT == 'true'
+      shell: bash
+      run: |
+        # Set key to use for JSON lookup
+        ACTOR="${{ github.actor }}"
+        ACTOR_ID="${{ github.actor_id }}"
+        echo "Using GitHub actor as lookup key: $ACTOR [$ACTOR_ID] 🔑"
+        echo "key=$ACTOR" >> "$GITHUB_ENV"
+
+    - name: "Get ticket from JSON lookup table"
+      if: env.PRESENT == 'false' && inputs.INJECT == 'true'
+      uses: lfit/releng-reusable-workflows/.github/actions/json-key-value-lookup-action@main
+      with:
+        json: ${{ inputs.issue_id_lookup_json }}
+        key: ${{ env.key }}
+
+    - name: "Configure GIT user for commit"
+      if: env.PRESENT == 'false' && inputs.INJECT == 'true'
+      uses: lfit/releng-reusable-workflows/.github/actions/git-configure-action@main
+
+    - name: "Adding JIRA ticket reference to commit message"
```

Comment:
Investigate force push closing pull request.

Reply:
Turns out the pull request "self closes" if the checkout is NOT performed with "fetch-depth: 0". I tried a value of 1 and that didn't work, either. I'm open to suggestions as to whether there are any better/alternative ways, apart from a force push, to do what I'm currently doing here?